File: Stack/Stack.hs
module Stack where

import Data.Maybe;

push :: [a] -> a -> [a]
push xs x = x:xs

pop :: [a] -> (Maybe a, [a])
pop [] = (Nothing, [])
pop (x:xs) = (Just x, xs)

back :: [a] -> Maybe a
back [] = Nothing
back (x:xs) = Just x

size :: [a] -> Int
size = length

clear :: [a] -> [a]
clear xs = []

exit :: [a] -> Int
exit xs = 0

execute :: (Show a, Eq a, Read a) => [String] -> [a] -> ([a], String, Int)
execute cmd stack =
    case (head cmd) of
        "push" -> ((push stack ((read . head . tail) cmd)), "ok", 1)
        "pop" ->
	    if((fst . pop) stack==Nothing)
	    then
	        ((snd . pop) stack, "error", 1)
	    else
	        ((snd . pop) stack, (show . fromJust . fst . pop) stack, 1)
        "back" ->
	    if(back stack==Nothing)
	    then
	        (stack, "error", 1)
	    else
	        (stack, (show . fromJust . back) stack, 1)
        "size" -> (stack, (show .size) stack, 1)
        "clear" -> (clear stack, "ok", 1)
        "exit" -> (stack, "bye", 0)

getAndExecuteCommand :: (Show a, Eq a, Read a) => [a] -> IO ()
getAndExecuteCommand xs = do
    command <- getLine
    printSecond (result command)
    if(getThird (result command)==1)
    then (getAndExecuteCommand . getFirst) (result command)
    else return ()
    where
        printSecond = putStrLn . getSecond
	getFirst (a, b, c) = a
	getSecond (a, b, c) = b
	getThird (a, b, c) = c
	result command = execute (words command) xs

nullIntList :: Int -> [Int]
nullIntList arg = []

main :: IO ()
main = (getAndExecuteCommand . nullIntList) 0

File: Stack/stack.py
def push(element, stack):
    stack.append(element)
    print("ok")

def pop(stack):
    if len(stack) > 0:
        print(stack[-1])
        stack.pop()
    else:
        print("error")

def back(stack):
    if len(stack) > 0:
        print(stack[-1])
    else:
        print("error")

def size(stack):
    print(len(stack))

def clear(stack):
    print("ok")
    stack.clear()

def exit():
    import sys
    print("bye")
    sys.exit(0)

def parsecommand(cmd, stack):
    if(cmd[0]=="push"):
        push(int(cmd[1]), stack)
    elif(cmd[0]=="pop"):
        pop(stack)
    elif(cmd[0]=="back"):
        back(stack)
    elif(cmd[0]=="size"):
        size(stack)
    elif(cmd[0]=="clear"):
        clear(stack)
    elif(cmd[0]=="exit"):
        exit()
    else:
        print("Unrecognized command")

stack = []
while 1>0:
    commandstr = input()
    parsecommand(commandstr.split(' '), stack)
